### PR TITLE
Tune defaults of Auto*Normal guides

### LINKED ISSUE #2120: Make autoguides more usable
This aims to make our most common autoguides easier to use: `AutoDelta`, `AutoDiagonalNormal`, `AutoMultivariateNormal`, `AutoLowRankMultivariateNormal`.

Feel free to add to this list (anyone :smile:) if you have usability requests.

I have added this to the 1.0 milestone because it involves interface changes that will enable later refactoring (e.g. #2078) while preserving backwards compatibility.

## Tasks
- [x] #2119 document use of custom `init_loc_fn`, and add examples in tutorials
- [x] #2119 add error checking logic in `init_loc_fn`
- [x] #2119 add easier `init_scale` args for `Auto*Normal` guides
- [x] #2126 make autoguides into `nn.Module`s so they are easier to `torch.load()`/`.save()`
    (this is an alternative to making the param store an `nn.Module` #2078 )
- [x] allow non-python-identifiers as site names in autoguides, ~~which was broken by #2126~~
- ~~make it easier to set optim configs per-guide-param (e.g. I often lower the learning rate `lr *= 0.1` for the `auto_cov_factor` param of `AutoLowRankMultivariateNormal`).~~
- [x] #2127 Stabilize parameterization of `AutoLowRankMultivariateNormal`
- [x] #2129 Support pickling of easyguides
- [x] #2137 Make `AutoGuideList` a bonafide `nn.ModuleList`
- [x] #2140 make easyguides into `nn.Modules`
- [x] #2148 Fix saving of `AutoGuides` in the param store
- [ ] #2156 Change default `init_scale` from 1.0 to 0.1 (@fritzo often uses this in practice)

### PR DESCRIPTION
**Pull request by fritzo**
Resolves #2120 

1. Reduce `init_scale` from 1.0 to 0.1. I typically use 0.1 for small models and 0.01 for large models. I choose 0.1 as a nice value for beginner users. Not so large that the model goes wild early in training (which I often see with 1.0), but not so small that it takes a long time for variance to increase.
2. Use a default heuristic of `rank ~ sqrt(latent_dim)` in `AutoLowRankMultivariateNormal`. This is a good computational tradeoff in practice. The old default of `rank=1` was silly.
3. Tune optimizers to reduce test time. `pytest -n auto tests/infer/test_autoguide.py` reduces from 42sec to 12sec on my laptop.

## Tested
- updated `test_mean` and `test_quantiles`
- used these defaults in small and large practical models

### CODE REVIEW

File: pyro/infer/autoguide/guides.py [hunk 1]
```diff
@@ -607,22 +605,24 @@ class AutoLowRankMultivariateNormal(AutoContinuous):
         guide = AutoLowRankMultivariateNormal(model, rank=10)
         svi = SVI(model, guide, ...)
 
-    By default the ``cov_diag`` is initialized to 1/2 and the ``cov_factor`` is
-    intialized randomly such that ``cov_factor.matmul(cov_factor.t())`` is half the
-    identity matrix.
+    By default the ``cov_diag`` is initialized to a small constant and the
+    ``cov_factor`` is intialized randomly such that on average
```
Comment (martinjankowiak) [inline pyro/infer/autoguide/guides.py]: typo
